Core/MMKV_IO.cpp

```diff
@@ -1352,9 +1355,12 @@ void MMKV::clearAll() {
 #else
     writeActualSize(0, 0, nullptr, IncreaseSequence);
 #endif
-    m_metaFile->msync(MMKV_SYNC);
 
-    clearMemoryCache();
+    if (!keepSpace) {
+        m_metaFile->msync(MMKV_SYNC);
```

Comment:
Now a second look at it, this `msync()` call on metafile is a **MUST** for we are `IncreaseSequence`.